swabbie-edda/src/main/kotlin/com/netflix/spinnaker/swabbie/edda/providers/EddaApiSupport.kt

```diff
@@ -29,9 +29,9 @@ open class EddaApiSupport(
 ) {
   private val log: Logger = LoggerFactory.getLogger(javaClass)
   val eddaFailureCountId: Id = registry.createId("swabbie.edda.failures")
-  fun withEddaClient(region: String, accountId: String): EddaService? {
+  fun withEddaClient(region: String, accountId: String, environment: String): EddaService? {
```

Comment:
This addresses an issue caused by a difference in the way clouddriver and edda represent AWS accounts partitioned into >1 environment. Environments are treated as distinct accounts by both clouddriver and edda, but they share the same AWS account id.

If AWS account spinnaker (id: 1234) was partitioned into test and prod envs, clouddriver would expose them as accounts spinnaker and spinnakertest. Edda would index them under instances edda-1234.test and edda-1234.prod. Each edda instance would only index their environment, not the full 1234 account.

Swabbie made an assumption that AWS account id's were unique; all resource providers fetched a single edda client looked up by account id and region. This could result in the incorrect edda instance being queried for partitioned accounts. Its also possible that some environments would have been excluded from edda lookups, i.e. both spinnaker and spinnakertest accounts could both potentially contain spinnakertest resources.

Reply:
Great find.